tools/utils/current_git.sh

```diff
@@ -0,0 +1,53 @@
+#!/bin/zsh
+
+# Usage:
+# $ ./tools/utils/current_git.sh -h 
+# Prints the current Git reference, either a tag or a branch name.
+
+# Options:
+#   --print: Outputs the current Git reference to STDOUT (tag if available, branch otherwise)
+#   --print-tag: Outputs the current Git tag to STDOUT (if available)
+#   --print-branch: Outputs the current Git branch to STDOUT
+
+set -eo pipefail
+
+# Prints current git tag (if any)
+function current_git_tag() {
+    if [[ -n "$CI_COMMIT_TAG" ]]; then
+        echo "$CI_COMMIT_TAG"
+    fi
+}
+
+# Prints current git branch
```

Comment:
/question

what happens when HEAD doesn't point to a branch?

Reply:
Like in detached HEAD configuration? It will print just `"HEAD"` in such case. Although, this code isn't new - we've been using it for years in existing smoke tests. I don't think GitLab setup changes anything significant to break it 🤞.